doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1836 @@
+systems is used and what the different assumptions predicates mean.
+
+.. note:: This page describes the "old assumptions" system. There is also a
+          "new assumptions" system which is described at the end of this page.
+
+Firstly we consider what happens when taking the square root of the square of
+a concrete integer such as $2$ or $-2$:
+
+    >>> from sympy import sqrt
+    >>> sqrt(2**2)
+    2
+    >>> sqrt((-2)**2)
+    2
+    >>> x = 2
+    >>> sqrt(x**2)
+    2
+    >>> sqrt(x**2) == x
+    True
+    >>> y = -2
+    >>> sqrt(y**2) == y
+    False
+    >>> sqrt(y**2) == -y
+    True
+
+What these examples demonstrate is that for a positive number $x$ we have
+$\sqrt{x^2} = x$ whereas for a negative number we would instead have
+$\sqrt{x^2} = -x$. That may seem obvious but the situation can be more
+surprising when working with a symbol rather then an explicit number. For
+example
+
+    >>> from sympy import Symbol, simplify
+    >>> x = Symbol('x')
+    >>> sqrt(x**2)
+    sqrt(x**2)
+
+It might look as if that should simplify to ``x`` but it does not even if
+:func:`~.simplify` is used:
+
+    >>> simplify(sqrt(x**2))
+    sqrt(x**2)
+
+This is because SymPy will refuse to simplify this expression if the
+simplification is not valid for *every* possible value of ``x``. By default
+the symbol ``x`` is considered only to represent something roughly like an
+arbitrary complex number and the obvious simplification here is only valid for
+positive real numbers. Since ``x`` is not known to be positive or even real no
+simplification of this expression is possible.
+
+We can tell SymPy that a symbol represents a positive real number when
+creating the symbol and then the simplification will happen automatically:
+
+    >>> y = Symbol('y', positive=True)
+    >>> sqrt(y**2)
+    y
+
+This is what is meant by "assumptions" in SymPy. If the symbol ``y``
+is created with ``positive=True`` then SymPy will *assume* that it represents
+a positive real number rather than an arbitrary complex or possibly infinite
+number. That *assumption* can make it possible to simplify expressions or
+might allow other manipulations to work. It is usually a good idea to be as
+precise as possible about the assumptions on a symbol when creating it.
+
+
+The (old) assumptions system
+============================
+
+There are two sides to the assumptions system. The first side is that we can
+declare assumptions on a symbol when creating the symbol. The other side is
+that we can query the assumptions on any expression using the corresponding
+``is_*`` attribute. For example:
+
+    >>> x = Symbol('x', positive=True)
+    >>> x.is_positive
+    True
+
+We can query assumptions on any expression not just a symbol:
+
+    >>> expr = 1 + x**2
+    >>> expr
+    x**2 + 1
+    >>> expr.is_positive
+    True
+    >>> expr.is_negative
+    False
+
+The values given in an assumptions query use three-valued "fuzzy" logic. Any
```

Comment:
It's a bit unfortunate that we call this "fuzzy" in functions like `fuzzy_and`, because fuzzy logic usually means something else https://en.wikipedia.org/wiki/Fuzzy_logic. Maybe a footnote should be here indicating that this is not to be confused with that.